# Feature engineering

In [1]:
# import library
import pandas as pd
import numpy as np

In [2]:
big_game = pd.read_excel("C://Users//AMAN//Downloads//biggame.xlsx")

In [3]:
big_game

Player Name Player Jersey Number Player Position  Player Age  \
0       Devin McCourty                  #32            S/FS          30   
1       Danny Amendola                  #80        PR/WR/KR          32   
2     Johnson Bademosi                  #29     CB/SPTM/RCB          27   
3          Chris Hogan                  #15              WR          29   
4        James Develin                  #46           RB/FB          29   
..                 ...                  ...             ...         ...   
113         Nick Foles                   #9              QB          28   
114        Will Beatty                  #66           T/LOT          32   
115       Donnie Jones                   #8           P/HLD          37   
116  Kamu Grugier-Hill                  #54     LB/SPTM/RLB          23   
117      Isaac Seumalo                  #73           G/ROG          24   

     Player Weight (lbs.) Years Played Player Birthplace (city, town, etc.)  \
0                     195            8                               Nanuet   
1                     190            9                        The Woodlands   
2                     206            6                        Silver Spring   
3                     210            5                              Wyckoff   
4                     255            5                        Gilbertsville   
..                    ...          ...                                  ...   
113                   243            6                               Austin   
114                   319            9                                 York   
115                   221           14                          Baton Rouge   
116                   220            2                             Honolulu   
117                   303            2                             Honolulu   

    Player Birth State    Player Birthplace (Combo)    Player College  ...  \
0             New York             Nanuet, New York           Rutgers  ...   
1                Texas         The Woodlands, Texas        Texas Tech  ...   
2             Maryland      Silver Spring, Maryland          Stanford  ...   
3           New Jersey          Wyckoff, New Jersey          Monmouth  ...   
4         Pennsylvania  Gilbertsville, Pennsylvania             Brown  ...   
..                 ...                          ...               ...  ...   
113              Texas                Austin, Texas           Arizona  ...   
114       Pennsylvania           York, Pennsylvania       Connecticut  ...   
115          Louisiana       Baton Rouge, Louisiana               LSU  ...   
116             Hawaii             Honolulu, Hawaii  Eastern Illinois  ...   
117             Hawaii             Honolulu, Hawaii      Oregon State  ...   

      Full GEOID Latitude (player birthplace) Longitude (player birthplace)  \
0    0400000US36                    41.088707                    -74.013473   
1    0400000US48                    30.173419                    -95.504686   
2    0400000US24                    38.990666                    -77.026088   
3    0400000US34                    41.009542                    -74.172922   
4    0400000US42                    40.320097                    -75.610184   
..           ...                          ...                           ...   
113  0400000US48                    30.271129                    -97.743699   
114  0400000US42                    39.962398                    -76.727392   
115  0400000US22                    30.450746                    -91.154551   
116  0400000US15                    21.304547                   -157.855676   
117  0400000US15                    21.304547                   -157.855676   

     Number from City Number of Records  \
0                   1                 1   
1                   1                 1   
2                   1                 1   
3                   1                 1   
4                   1                 1   
..               

In [4]:
big_game.head(5)

Player Name Player Jersey Number Player Position  Player Age  \
0    Devin McCourty                  #32            S/FS          30   
1    Danny Amendola                  #80        PR/WR/KR          32   
2  Johnson Bademosi                  #29     CB/SPTM/RCB          27   
3       Chris Hogan                  #15              WR          29   
4     James Develin                  #46           RB/FB          29   

   Player Weight (lbs.) Years Played Player Birthplace (city, town, etc.)  \
0                   195            8                               Nanuet   
1                   190            9                        The Woodlands   
2                   206            6                        Silver Spring   
3                   210            5                              Wyckoff   
4                   255            5                        Gilbertsville   

  Player Birth State    Player Birthplace (Combo) Player College  ...  \
0           New York             Nanuet, New York        Rutgers  ...   
1              Texas         The Woodlands, Texas     Texas Tech  ...   
2           Maryland      Silver Spring, Maryland       Stanford  ...   
3         New Jersey          Wyckoff, New Jersey       Monmouth  ...   
4       Pennsylvania  Gilbertsville, Pennsylvania          Brown  ...   

    Full GEOID Latitude (player birthplace) Longitude (player birthplace)  \
0  0400000US36                    41.088707                    -74.013473   
1  0400000US48                    30.173419                    -95.504686   
2  0400000US24                    38.990666                    -77.026088   
3  0400000US34                    41.009542                    -74.172922   
4  0400000US42                    40.320097                    -75.610184   

   Number from City Number of Records  \
0                 1                 1   
1                 1                 1   
2                 1                 1   
3                 1                 1   
4                 1                 1   

   American FactFinder Link for more Census data points  \
0  https://factfinder.census.gov/bkmk/cf/1.0/en/p...      
1  https://factfinder.census.gov/bkmk/cf/1.0/en/p...      
2  https://factfinder.census.gov/bkmk/cf/1.0/en/p...      
3  https://factfinder.census.gov/bkmk/cf/1.0/en/p...      
4  https://factfinder.census.gov/bkmk/cf/1.0/en/p...      

                                     Quickfacts Link  \
0  https://www.census.gov/quickfacts/fact/table/N...   
1  https://www.census.gov/quickfacts/fact/table/T...   
2  https://www.census.gov/quickfacts/fact/table/S...   
3  https://www.census.gov/quickfacts/fact/table/W...   
4  https://www.census.gov/quickfacts/fact/table/G...   

                                     State Data Link  \
0  https://factfinder.census.gov/bkmk/cf/1.0/en/s...   
1  https://factfinder.census.gov/bkmk/cf/1.0/en/s...   
2  https://factfinder.census.gov/bkmk/cf/1.0/en/s...   
3  https://factfinder.census.gov/bkmk/cf/1.0/en/s...   
4  https://factfinder.census.gov/bkmk/cf/1.0/en/s...   

                     Source (Population States 2017)  \
0  U.S. Census Bureau, 2017 Annual Estimates of t...   
1  U.S. Census Bureau, 2017 Annual Estimates of t...   
2  U.S. Census Bureau, 2017 Annual Estimates of t...   
3  U.S. Census Bureau, 2017 Annual Estimates of t...   
4  U.S. Census Bureau, 2017 Annual Estimates of t...   

                  Birthplace, Population Data Source  
0  U.S. Census Bureau, 2012-2016 American Communi...  
1  U.S. Census Bureau, 2012-2016 American Communi...  
2  U.S. Census Bureau, 2012-2016 American Communi...  
3  U.S. Census Bureau, 2012-2016 American Communi...  
4  U.S. Census Bureau, 2012-2016 American Communi...  

[5 rows x 24 columns]

In [5]:
big_game.tail(5)

Player Name Player Jersey Number Player Position  Player Age  \
113         Nick Foles                   #9              QB          28   
114        Will Beatty                  #66           T/LOT          32   
115       Donnie Jones                   #8           P/HLD          37   
116  Kamu Grugier-Hill                  #54     LB/SPTM/RLB          23   
117      Isaac Seumalo                  #73           G/ROG          24   

     Player Weight (lbs.) Years Played Player Birthplace (city, town, etc.)  \
113                   243            6                               Austin   
114                   319            9                                 York   
115                   221           14                          Baton Rouge   
116                   220            2                             Honolulu   
117                   303            2                             Honolulu   

    Player Birth State Player Birthplace (Combo)    Player College  ...  \
113              Texas             Austin, Texas           Arizona  ...   
114       Pennsylvania        York, Pennsylvania       Connecticut  ...   
115          Louisiana    Baton Rouge, Louisiana               LSU  ...   
116             Hawaii          Honolulu, Hawaii  Eastern Illinois  ...   
117             Hawaii          Honolulu, Hawaii      Oregon State  ...   

      Full GEOID Latitude (player birthplace) Longitude (player birthplace)  \
113  0400000US48                    30.271129                    -97.743699   
114  0400000US42                    39.962398                    -76.727392   
115  0400000US22                    30.450746                    -91.154551   
116  0400000US15                    21.304547                   -157.855676   
117  0400000US15                    21.304547                   -157.855676   

     Number from City Number of Records  \
113                 1                 1   
114                 1                 1   
115                 2                 1   
116                 2                 1   
117                 2                 1   

     American FactFinder Link for more Census data points  \
113  https://factfinder.census.gov/bkmk/cf/1.0/en/p...      
114  https://factfinder.census.gov/bkmk/cf/1.0/en/p...      
115  https://factfinder.census.gov/bkmk/cf/1.0/en/p...      
116  https://factfinder.census.gov/bkmk/cf/1.0/en/p...      
117  https://factfinder.census.gov/bkmk/cf/1.0/en/p...      

                                       Quickfacts Link  \
113  https://www.census.gov/quickfacts/fact/table/A...   
114  https://www.census.gov/quickfacts/fact/table/Y...   
115  https://www.census.gov/quickfacts/fact/table/B...   
116  https://www.census.gov/quickfacts/fact/table/H...   
117  https://www.census.gov/quickfacts/fact/table/H...   

                                       State Data Link  \
113  https://factfinder.census.gov/bkmk/cf/1.0/en/s...   
114  https://factfinder.census.gov/bkmk/cf/1.0/en/s...   
115  https://factfinder.census.gov/bkmk/cf/1.0/en/s...   
116  https://factfinder.census.gov/bkmk/cf/1.0/en/s...   
117  https://factfinder.census.gov/bkmk/cf/1.0/en/s...   

                       Source (Population States 2017)  \
113  U.S. Census Bureau, 2017 Annual Estimates of t...   
114  U.S. Census Bureau, 2017 Annual Estimates of t...   
115  U.S. Census Bureau, 2017 Annual Estimates of t...   
116  U.S. Census Bureau, 2017 Annual Estimates of t...   
117  U.S. Census Bureau, 2017 Annual Estimates of t...   

                    Birthplace, Population Data Source  
113  U.S. Census Bureau, 2016 Annual Estimates of t...  
114  U.S. Census Bureau, 2016 Annual Estimates of t...  
115  U.S. Census Bureau, 2016 Annual Estimates of t...  
116  U.S. Census Bureau, 2016 Annual Estimates of t...  
117  U.S. Census Bureau, 2016 Annual Estimates of t...  

[5 rows x 24 columns]

In [6]:
big_game.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 118 entries, 0 to 117
Data columns (total 24 columns):
 #   Column                                                    Non-Null Count  Dtype  
---  ------                                                    --------------  -----  
 0   Player Name                                               118 non-null    object 
 1   Player Jersey Number                                      118 non-null    object 
 2   Player Position                                           118 non-null    object 
 3   Player Age                                                118 non-null    int64  
 4   Player Weight (lbs.)                                      118 non-null    int64  
 5   Years Played                                              118 non-null    object 
 6   Player Birthplace (city, town, etc.)                      118 non-null    object 
 7   Player Birth State                                        118 non-null    object 
 8   Player Birthplace (C

In [6]:
big_game["Player Age"].mean()

26.720338983050848

In [7]:
big_game["Player Weight (lbs.)"].mean()

243.88135593220338

In [8]:
big_game.describe()

Player Age  Player Weight (lbs.)  State GEO ID  \
count  118.000000            118.000000    118.000000   
mean    26.720339            243.881356     28.330508   
std      3.653556             45.959322     17.071521   
min     21.000000            167.000000      1.000000   
25%     24.000000            205.000000     12.000000   
50%     26.000000            238.000000     31.000000   
75%     29.000000            274.000000     40.500000   
max     40.000000            350.000000     75.000000   

       Latitude (player birthplace)  Longitude (player birthplace)  \
count                    118.000000                     118.000000   
mean                      35.566567                     -92.820792   
std                        7.449830                      20.707252   
min                      -14.252294                    -170.713450   
25%                       32.416135                     -97.764743   
50%                       36.164258                     -86.614908   
75%                       40.411295                     -80.155949   
max                       51.507322                      -0.127647   

       Number from City  Number of Records  
count        118.000000              118.0  
mean           1.152542                1.0  
std            0.426216                0.0  
min            1.000000                1.0  
25%            1.000000                1.0  
50%            1.000000                1.0  
75%            1.000000                1.0  
max            3.000000                1.0

# Form Age of player we can entersapt that 

#### min age of player is 21 years

#### and max age of player is 40 years

#### avg age of a player is 26 years

#### 75% of player belong to age group between 24-29

# Form Player Weight (lbs.) of player we can entersapt that

#### min weight of player is 167 lbs

#### max wegiht of player is 350 lbs

#### avg weight of player is 243.88lbs

#### 75% of player have weight between 167-274 lbs

In [9]:
#  what are unique values in year player
big_game['Years Played'].unique()

array([8, 9, 6, 5, 4, 2, 3, 1, 10, 15, 12, 11, 'R', 7, 18, 13, 14],
      dtype=object)

## the most experianced player has 18 year of experiance and atleast one year experiance

In [10]:
big_game['Player Team'].describe(include=['O'])

count             118
unique              2
top       New England
freq               60
Name: Player Team, dtype: object

In [11]:
big_game['Player Team'].unique()

array(['New England', 'Philadelphia'], dtype=object)

## Two Teams from which player from New England', 'Philadelphia

In [12]:
# what are different playing position
big_game['Player Position'].unique()

array(['S/FS', 'PR/WR/KR', 'CB/SPTM/RCB', 'WR', 'RB/FB', 'T/ROT', 'S/SS',
       'CB/LCB', 'WR/PR', 'RB', 'RB/KR', 'LB/RLB', 'KR/WR', 'LB/RDE',
       'C/ROG', 'CB/RCB', 'K/KFG/KKO', 'DT/RDT', 'DE/RDE', 'S/FS/SPTM',
       'LB/MLB', 'P/HLD', 'QB', 'DT', 'LB/LS', 'C', 'LB/LLB', 'TE',
       'DT/LDE', 'DE', 'G/C', 'DE/LDE', 'DE/LDT', 'T/LOG', 'G/LOG',
       'T/LOT', 'DT/LDT', 'RB/SPTM', 'S/SS/SPTM', 'KR/RB', 'RB/KR/PR',
       'LB/LLB/SPTM', 'C/LOG', 'G/ROG', 'LB/SPTM/RLB'], dtype=object)

# We look into the  2017 population data of states

In [13]:
state=pd.read_excel("C://Users//AMAN//Downloads//All states Census 2017 Population Estimates.xlsx")

In [14]:
state.head()

GEOID  GEOID2 Geography Name  April 1, 2010 - Census  \
0  0400000US01       1        Alabama                 4779736   
1  0400000US02       2         Alaska                  710231   
2  0400000US04       4        Arizona                 6392017   
3  0400000US05       5       Arkansas                 2915918   
4  0400000US06       6     California                37253956   

   April 1, 2010 - Estimates Base  Population Estimate (as of July 1) - 2010  \
0                         4780135                                    4785579   
1                          710249                                     714015   
2                         6392309                                    6407002   
3                         2916031                                    2921737   
4                        37254518                                   37327690   

   Population Estimate (as of July 1) - 2011  \
0                                    4798649   
1                                     722259   
2                                    6465488   
3                                    2938640   
4                                   37672654   

   Population Estimate (as of July 1) - 2012  \
0                                    4813946   
1                                     730825   
2                                    6544211   
3                                    2949208   
4                                   38019006   

   Population Estimate (as of July 1) - 2013  \
0                                    4827660   
1                                     736760   
2                                    6616124   
3                                    2956780   
4                                   38347383   

   Population Estimate (as of July 1) - 2014  \
0                                    4840037   
1                                     736759   
2                                    6706435   
3                                    2964800   
4                                   38701278   

   Population Estimate (as of July 1) - 2015  \
0                                    4850858   
1                                     737979   
2                                    6802262   
3                                    2975626   
4                                   39032444   

   Population Estimate (as of July 1) - 2016  \
0                                    4860545   
1                                     741522   
2                                    6908642   
3                                    2988231   
4                                   39296476   

   Population Estimate (as of July 1) - 2017  
0                                    4874747  
1                                     739795  
2                                    7016270  
3                                    3004279  
4                                   39536653

In [15]:
state.tail()

GEOID  GEOID2 Geography Name  April 1, 2010 - Census  \
47  0400000US53      53     Washington                 6724540   
48  0400000US54      54  West Virginia                 1852994   
49  0400000US55      55      Wisconsin                 5686986   
50  0400000US56      56        Wyoming                  563626   
51  0400000US72      72    Puerto Rico                 3725789   

    April 1, 2010 - Estimates Base  Population Estimate (as of July 1) - 2010  \
47                         6724545                                    6741386   
48                         1853006                                    1854315   
49                         5687288                                    5690403   
50                          563767                                     564376   
51                         3726157                                    3721525   

    Population Estimate (as of July 1) - 2011  \
47                                    6819155   
48                                    1854891   
49                                    5705812   
50                                     567602   
51                                    3678732   

    Population Estimate (as of July 1) - 2012  \
47                                    6890899   
48                                    1855360   
49                                    5721075   
50                                     576608   
51                                    3634488   

    Population Estimate (as of July 1) - 2013  \
47                                    6963410   
48                                    1852333   
49                                    5736673   
50                                     582341   
51                                    3593077   

    Population Estimate (as of July 1) - 2014  \
47                                    7046931   
48                                    1847624   
49                                    5751272   
50                                     583334   
51                                    3534874   

    Population Estimate (as of July 1) - 2015  \
47                                    7152818   
48                                    1839767   
49                                    5759744   
50                                     586102   
51                                    3473177   

    Population Estimate (as of July 1) - 2016  \
47                                    7280934   
48                                    1828637   
49                                    5772917   
50                                     584910   
51                                    3406520   

    Population Estimate (as of July 1) - 2017  
47                                    7405743  
48                                    1815857  
49                                    5795483  
50                                     579315  
51                                    3337177

In [16]:
state.info

<bound method DataFrame.info of           GEOID  GEOID2        Geography Name  April 1, 2010 - Census  \
0   0400000US01       1               Alabama                 4779736   
1   0400000US02       2                Alaska                  710231   
2   0400000US04       4               Arizona                 6392017   
3   0400000US05       5              Arkansas                 2915918   
4   0400000US06       6            California                37253956   
5   0400000US08       8              Colorado                 5029196   
6   0400000US09       9           Connecticut                 3574097   
7   0400000US10      10              Delaware                  897934   
8   0400000US11      11  District of Columbia                  601723   
9   0400000US12      12               Florida                18801310   
10  0400000US13      13               Georgia                 9687653   
11  0400000US15      15                Hawaii                 1360301   
12  0400000US16    

In [17]:
state.describe()

GEOID2  April 1, 2010 - Census  April 1, 2010 - Estimates Base  \
count  52.000000            5.200000e+01                    5.200000e+01   
mean   29.788462            6.009064e+06                    6.009313e+06   
std    16.774557            6.764460e+06                    6.764684e+06   
min     1.000000            5.636260e+05                    5.637670e+05   
25%    16.750000            1.761651e+06                    1.761658e+06   
50%    29.500000            4.085220e+06                    4.085206e+06   
75%    42.500000            6.591857e+06                    6.591992e+06   
max    72.000000            3.725396e+07                    3.725452e+07   

       Population Estimate (as of July 1) - 2010  \
count                               5.200000e+01   
mean                                6.020384e+06   
std                                 6.779468e+06   
min                                 5.643760e+05   
25%                                 1.765195e+06   
50%                                 4.092510e+06   
75%                                 6.609054e+06   
max                                 3.732769e+07   

       Population Estimate (as of July 1) - 2011  \
count                               5.200000e+01   
mean                                6.063904e+06   
std                                 6.848039e+06   
min                                 5.676020e+05   
25%                                 1.777026e+06   
50%                                 4.117175e+06   
75%                                 6.663922e+06   
max                                 3.767265e+07   

       Population Estimate (as of July 1) - 2012  \
count                               5.200000e+01   
mean                                6.108226e+06   
std                                 6.917273e+06   
min                                 5.766080e+05   
25%                                 1.789815e+06   
50%                                 4.138796e+06   
75%                                 6.717445e+06   
max                                 3.801901e+07   

       Population Estimate (as of July 1) - 2013  \
count                               5.200000e+01   
mean                                6.150530e+06   
std                                 6.982359e+06   
min                                 5.823410e+05   
25%                                 1.791796e+06   
50%                                 4.159392e+06   
75%                                 6.774206e+06   
max                                 3.834738e+07   

       Population Estimate (as of July 1) - 2014  \
count                               5.200000e+01   
mean                                6.195335e+06   
std                                 7.056337e+06   
min                                 5.833340e+05   
25%                                 1.793316e+06   
50%                                 4.185544e+06   
75%                                 6.830176e+06   
max                                 3.870128e+07   

       Population Estimate (as of July 1) - 2015  \
count                               5.200000e+01   
mean                                6.240635e+06   
std                                 7.131146e+06   
min                                 5.861020e+05   
25%                                 1.792156e+06   
50%                                 4.219297e+06   
75%                                 6.889901e+06   
max                                 3.903244e+07   

       Population Estimate (as of July 1) - 2016  \
count                               5.200000e+01   
mean                                6.284855e+06   
std                                 7.197166e+06   
min                                 5.849100e+05   
25%                                 1.791484e+06   
50%                                 4.261051e+06   
75%                                 7.001715e+06   
max                                 3.929648e+07   

       Population Estimate (as of July 1) 

In [18]:
state['Geography Name'].unique()

array(['Alabama', 'Alaska', 'Arizona', 'Arkansas', 'California',
       'Colorado', 'Connecticut', 'Delaware', 'District of Columbia',
       'Florida', 'Georgia', 'Hawaii', 'Idaho', 'Illinois', 'Indiana',
       'Iowa', 'Kansas', 'Kentucky', 'Louisiana', 'Maine', 'Maryland',
       'Massachusetts', 'Michigan', 'Minnesota', 'Mississippi',
       'Missouri', 'Montana', 'Nebraska', 'Nevada', 'New Hampshire',
       'New Jersey', 'New Mexico', 'New York', 'North Carolina',
       'North Dakota', 'Ohio', 'Oklahoma', 'Oregon', 'Pennsylvania',
       'Rhode Island', 'South Carolina', 'South Dakota', 'Tennessee',
       'Texas', 'Utah', 'Vermont', 'Virginia', 'Washington',
       'West Virginia', 'Wisconsin', 'Wyoming', 'Puerto Rico'],
      dtype=object)

In [19]:
state['GEOID2'].mean()

29.78846153846154

In [20]:
state['GEOID2'].max()

72

In [21]:
state['GEOID2'].median()

29.5

#  we look into the 2016 population data of cities

In [22]:
cities=pd.read_excel("C://Users//AMAN//Downloads//All Places Census 2016 Population Estimates.xlsx")

In [23]:
cities.head()

Geographic ID  GEOID 2 Geography, full name (City, State)  \
0  1620000US0100124   100124            Abbeville city, Alabama   
1  1620000US0100460   100460           Adamsville city, Alabama   
2  1620000US0100484   100484              Addison town, Alabama   
3  1620000US0100676   100676                Akron town, Alabama   
4  1620000US0100820   100820            Alabaster city, Alabama   

  April 1, 2010 - Census  April 1, 2010 - Estimates Base  \
0                   2688                            2688   
1                   4522                            4522   
2                    758                             756   
3                    356                             356   
4                  30352                           31066   

   Population Estimate (as of July 1) - 2010  \
0                                       2683   
1                                       4517   
2                                        754   
3                                        355   
4                                      31176   

   Population Estimate (as of July 1) - 2011  \
0                                       2685   
1                                       4495   
2                                        753   
3                                        345   
4                                      31362   

   Population Estimate (as of July 1) - 2012  \
0                                       2647   
1                                       4472   
2                                        748   
3                                        345   
4                                      31663   

   Population Estimate (as of July 1) - 2013  \
0                                       2631   
1                                       4447   
2                                        748   
3                                        342   
4                                      31960   

   Population Estimate (as of July 1) - 2014  \
0                                       2619   
1                                       4428   
2                                        747   
3                                        337   
4                                      32167   

   Population Estimate (as of July 1) - 2015  \
0                                       2616   
1                                       4395   
2                                        740   
3                                        337   
4                                      32751   

   Population Estimate (as of July 1) - 2016  
0                                       2603  
1                                       4360  
2                                        738  
3                                        334  
4                                      32948

In [24]:
cities.tail()

Geographic ID  GEOID 2 Geography, full name (City, State)  \
19505  1620000US5681300  5681300            Wamsutter town, Wyoming   
19506  1620000US5683040  5683040            Wheatland town, Wyoming   
19507  1620000US5684925  5684925              Worland city, Wyoming   
19508  1620000US5685015  5685015               Wright town, Wyoming   
19509  1620000US5686665  5686665                Yoder town, Wyoming   

      April 1, 2010 - Census  April 1, 2010 - Estimates Base  \
19505                    451                             451   
19506                   3627                            3627   
19507                   5487                            5487   
19508                   1807                            1807   
19509                    151                             151   

       Population Estimate (as of July 1) - 2010  \
19505                                        450   
19506                                       3629   
19507                                       5494   
19508                                       1807   
19509                                        152   

       Population Estimate (as of July 1) - 2011  \
19505                                        451   
19506                                       3625   
19507                                       5448   
19508                                       1809   
19509                                        154   

       Population Estimate (as of July 1) - 2012  \
19505                                        461   
19506                                       3634   
19507                                       5439   
19508                                       1855   
19509                                        155   

       Population Estimate (as of July 1) - 2013  \
19505                                        483   
19506                                       3625   
19507                                       5441   
19508                                       1850   
19509                                        159   

       Population Estimate (as of July 1) - 2014  \
19505                                        503   
19506                                       3650   
19507                                       5356   
19508                                       1844   
19509                                        162   

       Population Estimate (as of July 1) - 2015  \
19505                                        494   
19506                                       3658   
19507                                       5376   
19508                                       1866   
19509                                        162   

       Population Estimate (as of July 1) - 2016  
19505                                        484  
19506                                       3606  
19507                                       5316  
19508                                       1834  
19509                                        159

In [25]:
cities.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19510 entries, 0 to 19509
Data columns (total 12 columns):
 #   Column                                     Non-Null Count  Dtype 
---  ------                                     --------------  ----- 
 0   Geographic ID                              19510 non-null  object
 1   GEOID 2                                    19510 non-null  int64 
 2   Geography, full name (City, State)         19510 non-null  object
 3   April 1, 2010 - Census                     19510 non-null  object
 4   April 1, 2010 - Estimates Base             19510 non-null  int64 
 5   Population Estimate (as of July 1) - 2010  19510 non-null  int64 
 6   Population Estimate (as of July 1) - 2011  19510 non-null  int64 
 7   Population Estimate (as of July 1) - 2012  19510 non-null  int64 
 8   Population Estimate (as of July 1) - 2013  19510 non-null  int64 
 9   Population Estimate (as of July 1) - 2014  19510 non-null  int64 
 10  Population Estimate (as of July 1)

In [26]:
cities.describe()

GEOID 2  April 1, 2010 - Estimates Base  \
count  1.951000e+04                    1.951000e+04   
mean   3.008271e+06                    9.909960e+03   
std    1.461650e+06                    8.012455e+04   
min    1.001240e+05                    0.000000e+00   
25%    1.857141e+06                    3.690000e+02   
50%    2.940709e+06                    1.147000e+03   
75%    4.202612e+06                    4.603250e+03   
max    5.686665e+06                    8.174962e+06   

       Population Estimate (as of July 1) - 2010  \
count                               1.951000e+04   
mean                                9.930210e+03   
std                                 8.028673e+04   
min                                 0.000000e+00   
25%                                 3.690000e+02   
50%                                 1.148000e+03   
75%                                 4.606000e+03   
max                                 8.192026e+06   

       Population Estimate (as of July 1) - 2011  \
count                               1.951000e+04   
mean                                1.001311e+04   
std                                 8.113065e+04   
min                                 0.000000e+00   
25%                                 3.682500e+02   
50%                                 1.148000e+03   
75%                                 4.620000e+03   
max                                 8.284098e+06   

       Population Estimate (as of July 1) - 2012  \
count                               1.951000e+04   
mean                                1.009960e+04   
std                                 8.195162e+04   
min                                 0.000000e+00   
25%                                 3.660000e+02   
50%                                 1.147000e+03   
75%                                 4.632500e+03   
max                                 8.361179e+06   

       Population Estimate (as of July 1) - 2013  \
count                               1.951000e+04   
mean                                1.018003e+04   
std                                 8.265639e+04   
min                                 0.000000e+00   
25%                                 3.662500e+02   
50%                                 1.146500e+03   
75%                                 4.648000e+03   
max                                 8.422460e+06   

       Population Estimate (as of July 1) - 2014  \
count                               1.951000e+04   
mean                                1.026316e+04   
std                                 8.329606e+04   
min                                 0.000000e+00   
25%                                 3.650000e+02   
50%                                 1.145500e+03   
75%                                 4.680000e+03   
max                                 8.471990e+06   

       Population Estimate (as of July 1) - 2015  \
count                               1.951000e+04   
mean                                1.034566e+04   
std                                 8.389854e+04   
min                                 0.000000e+00   
25%                                 3.650000e+02   
50%                                 1.146500e+03   
75%                                 4.700500e+03   
max                                 8.516502e+06   

       Population Estimate (as of July 1) - 2016  
count                               1.951000e+04  
mean                                1.042104e+04  
std                                 8.432409e+04  
min                                 0.000000e+00  
25%                                 3.640000e+02  
50%                                 1.149500e+03  
75%                                 4.716500e+03  
max                                 8.537673e+06

In [27]:
cities['GEOID 2'].mean()

3008271.2834443874

In [28]:
cities['GEOID 2'].median()

2940709.0

In [29]:
cities['GEOID 2'].max()

5686665

# END